scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -861,6 +862,22 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
     }
 
   /**
+   * Save this SCollection as a Protobuf file.
+   * @group output
+   */
+  def saveAsProtobufFile(path: String, numShards: Int = 0)
+                        (implicit ev: T <:< GeneratedMessage with Message): Future[Tap[T]] =
+    if(context.isTest) {
+      context.testOut(ProtobufIO(path))(this)
+      saveAsInMemoryTap
+    } else {
+      this.asInstanceOf[SCollection[Message]]
+        .map(_.toByteArray)
+        .saveAsObjectFile(path, numShards)
```

Comment:
`.saveAsObjectFile` will still serialize the `Array[Byte]` using Kryo, is that transparent?
Or we can add private `ScioContext#AvroBytesFile` & `SCollection#saveAsAvroBytesFile` and use them for both object file and proto file?